File: fibonacci/fibonacci.java

public class fibonacci {
  public static void main(String[] args) {
    
    int i;

    /* If there's an argument, fib it! */
    if (args.length == 1) {
      fib(Integer.parseInt(args[0]));
      System.out.println();
      return;
    }

    /* Else, run 20 fibs */
    for (i = 0; i < 10; i++) {
      fib(i);
      System.out.print("\t");
      fib(i+10);
      System.out.println();
    }
    return;
  }

  private static void fib(int rotations) {

    int tmp;
    int old_value = 1;
    int new_value = 0;

    while (rotations > 0) {
      tmp = old_value;
      old_value = new_value;
      new_value = tmp + old_value;
      rotations -= 1;
    }
    
    System.out.print(new_value);
  }
}

/* TAIL INFO:
Name: Fibonacci Sequence
Language: Java
Compile: javac fibonacci.java
State: Done

Prints out numbers from the fibonacci sequence

Example: java fibonacci
Example2: java fibonacci 42
*/

File: fibonacci/fibonacci.cpp
#include <iostream>
#include <cstdlib>

unsigned long fib (int i) 
{
  unsigned long oldval = 1;
  unsigned long newval = 0;
  unsigned long tmp;

  while (--i >= 0)
  {
    tmp = oldval;
    oldval = newval;
    newval = tmp + oldval;
  }
  return newval;
}

int main (int argc, char **argv) 
{

  // Fib the first argument, if possible
  if (argc == 2) 
  {
    std::cout << fib(atoi(argv[1])) << std::endl;
    return 0;
  }

  // Otherwise, run 20 fibs
  for (int i = 0; i < 10; i++)
    std::cout << fib(i) << "\t" << fib(i+10) << std::endl;
  return 0;
}

// TAIL INFO:
// Name: Fibonacci Sequence
// Language: C++
// Compile: g++ fibonacci.cpp -o fibonacci_cpp
// State: Done
// Created: 2013-08-08
//
// Prints out numbers from the fibonacci sequence
// 
// Example1: ./fibonacci_cpp
// Example2: ./fibonacci_cpp 42
